runtime/src/main/java/org/corfudb/runtime/collections/CorfuQueue.java

```diff
@@ -181,8 +190,8 @@ public String toString() {
             return String.format("%s=>%s", recordId, entry);
         }
 
-        CorfuQueueRecord(long ordering, long entryId, E entry) {
-            this.recordId = new CorfuRecordId(ordering, entryId);
+        CorfuQueueRecord(long epoch, long ordering, long entryId, E entry) {
```

Comment:
why not to pass `CorfuRecordId` as a parameter?

Reply:
because these fields need to be accessed and set individually at different times I thought it would be cleaner to separate those two.

Reply:
ok